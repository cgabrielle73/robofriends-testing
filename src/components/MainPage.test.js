
import { shallow } from "enzyme";
import React from 'react';
import MainPage from "./MainPage";

let wrapper;

beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps}/>)
});

it('renders main page without crashing', () => {
    expect(wrapper).toMatchSnapshot();
});

it('filters robots', () => {
    expect(wrapper.instance().filterRobots()).toEqual([]);
});

it('filters robots2', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 1,
            name: 'Clara',
            email: 'mock@gmail.com'
        }],
        searchField: 'clara',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2}/>)
    expect(wrapper2.instance().filterRobots()).toEqual([{
        id: 1,
        name: 'Clara',
        email: 'mock@gmail.com'
    }]);
});

it('filters robots2', () => {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 1,
            name: 'Clara',
            email: 'mock@gmail.com'
        }],
        searchField: 'b',
        isPending: false
    }
    const filteredRobots = []
    const wrapper3 = shallow(<MainPage {...mockProps3}/>)
    expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
});

it('isPending robots2', () => {
    const mockProps4 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 1,
            name: 'Clara',
            email: 'mock@gmail.com'
        }],
        searchField: '',
        isPending: true
    }
    const wrapper4 = shallow(<MainPage {...mockProps4}/>)
    expect(wrapper4.instance().props.isPending).toEqual(true);
});
