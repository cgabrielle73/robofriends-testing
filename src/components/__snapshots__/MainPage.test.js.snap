// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders main page without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
